paasta_tools/cli/cmds/remote_run.py

```diff
@@ -0,0 +1,314 @@
+    )
+    list_parser.add_argument(
+        '--json-dict',
+        help='When running dry run, output the arguments as a json dict',
+        action='store_true',
+        dest='dry_run_json_dict',
+    )
+    list_parser.add_argument(
+        '-C', '--cmd',
+        help=('Run Docker container with particular command, '
+              'for example: "bash". By default will use the command or args specified by the '
+              'soa-configs or what was specified in the Dockerfile'),
+        required=False,
+        default=None,
+    )
+    list_parser.add_argument(
+        '-i', '--instance',
+        help=("Simulate a docker run for a particular instance of the service, like 'main' or 'canary'"),
+        required=False,
+        default=None,
+    ).completer = lazy_choices_completer(list_instances)
+    list_parser.add_argument(
+        '-v', '--verbose',
+        help='Show Docker commands output',
+        action='store_true',
+        required=False,
+        default=True,
+    )
+    list_parser.add_argument(
+        '-d', '--dry-run',
+        help='Don\'t launch the task',
+        action='store_true',
+        required=False,
+        default=False,
+    )
+    list_parser.add_argument(
+        '-l', '--local',
+        help='Run mesos-related bits locally, rather than ssh\'ing to mesos-master',
+        action='store_true',
+        required=False,
+        default=False,
+    )
+    list_parser.set_defaults(command=paasta_remote_run)
+
+
+class UnknownPaastaRemoteServiceError(Exception):
+    pass
+
+
+def make_config_reader(instance_type):
+    def reader(service, instance, cluster, soa_dir=DEFAULT_SOA_DIR):
+        conf_file = '%s-%s' % (instance_type, cluster)
+        full_path = '%s/%s/%s.yaml' % (soa_dir, service, conf_file)
+        paasta_print("Reading paasta-remote configuration file: %s" % full_path)
+        config = service_configuration_lib.read_extra_service_information(service, conf_file, soa_dir=soa_dir)
+
+        if instance not in config:
+            raise UnknownPaastaRemoteServiceError(
+                'No job named "%s" in config file %s: \n%s' % (instance, full_path, open(full_path).read())
+            )
+
+        return config
+
+    return reader
+
+
+def run_framework(args, system_paasta_config):
+    service = figure_out_service_name(args, soa_dir=args.yelpsoa_config_root)
+    cluster = args.cluster or system_paasta_config.get_local_run_config().get('default_cluster', None)
+
+    if not cluster:
+        paasta_print(
+            PaastaColors.red(
+                "PaaSTA on this machine has not been configured with a default cluster."
+                "Please pass one using '-c'."),
+            sep='\n',
+            file=sys.stderr,
+        )
+        return 1
+
+    soa_dir = args.yelpsoa_config_root
+    dry_run = args.dry_run
+    instance = args.instance
+    command = shlex.split(args.cmd, posix=False) if args.cmd else None
+
+    if instance is None:
```

Comment:
seemed like sensible defaults. `adhoc-%s.yaml` has `interactive` section for local-run, let's have `remote` for `remote-run`?